common/appscale/common/deployment_config.py

```diff
@@ -29,24 +27,56 @@ class ConfigStates(object):
   LOADED = 'loaded'
 
 
+class DeploymentConfigSection(object):
+  """ Keeps track of a section of configuration data. """
+  def __init__(self, zk_client, section):
+    """ Creates a new DeploymentConfigSection.
+
+    Args:
+      zk_client: A KazooClient.
+      section: A string specifying a configuration section name.
+    """
+    self.logger = logging.getLogger(self.__class__.__name__)
+    self.section_name = section
+    self.data = {}
+
+    section_node = '/appscale/config/{}'.format(section)
+    self.watch = zk_client.DataWatch(section_node, self._update_section)
+
+  def stop(self):
+    """ Stops the DataWatch on the configuration section. """
+    self.watch._stopped = True
```

Comment:
That's a great point about using private attributes, and it also made me realize that I was leaving around unnecessary "exists" watches. However, the vaguely-named `remove_listener` actually just removes the connection state listener and does not affect the watch.

You can't really stop a watch without closing the connection. You can only prevent it from being reset after it fires.

The reason I was using `_stopped` instead of returning `False` from the handler was because I was worried about the slim possibility of this happening:
```
[child 1 exists in ZK]
[child 2 is added]
ChildrenWatch fires: child1, child2
DataWatch (child 2) fires
DataWatch (child 2) reset
[child 2 is deleted from ZK]
DataWatch (child 2) fires (and is not reset)
[child 2 is re-added]
ChildrenWatch reset
ChildrenWatch fires: child 1, child 2
```
With the above timeline, the handler on the last ChildrenWatch fire does not see any changes, so it will not recreate the DataWatch for child 2.

With my latest commit, I addressed that by making the ChildrenWatch handlers ensure that all of the relevant DataWatches are active.